src/functions/types.ts

```diff
@@ -101,21 +109,24 @@ type FunctionInputRuntimeType<
 type UnknownRuntimeType = any;
 
 type TypedObjectFunctionInputRuntimeType<
-  Param extends TypedObjectParameterDefinition,
-> = Param["additionalProperties"] extends false ? {
-    [k in keyof Param["properties"]]: FunctionInputRuntimeType<
-      Param["properties"][k]
+  Props extends TypedObjectProperties,
```

Comment:
This is the main change enabling proper object sub-property optionality autocomplete in a function implementation context.